Automattic/wp-calypso
# wpcom-proxy-request: Return Promise if no callback specified (Take 3)

Reverts Automattic/wp-calypso#39823
Take 3 of https://github.com/Automattic/wp-calypso/pull/39683:

> #### Changes proposed in this Pull Request
> The promise wrapper introduced by @andrewserong et al. in #38823 seems generic enough for it to make sense to have it in `wpcom-proxy-request`. This PR moves it there. This follows [precedent](https://github.com/Automattic/wp-calypso/blob/bf345c5b040e75808fb2f92c9b9e34e6f126a235/packages/wpcom.js/src/lib/util/send-request.js#L78-L93) set by `wpcom.js`.
> 
> Note that this is a technically a breaking change, as noted in the package's `History.md`:
> 
> > * [...] Return Promise (rather than `XMLHttpRequest` instance) if no callback argument is provided.
> >   
> >   * In practice, most people have probably been using the callback rather than the returned `XMLHttpRequest` instance, so this shouldn't be a breaking change for most.
> 
> #### Testing instructions
> Verify that Gutenboarding still works (esp account and site creation).



Conflicting changes had been merged in the interim. I've updated the imports to fix the issue from  https://github.com/Automattic/wp-calypso/pull/39722
